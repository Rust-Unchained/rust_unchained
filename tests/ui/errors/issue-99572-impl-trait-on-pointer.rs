// Emit additional suggestion to correct the trait implementation
// on a pointer
use std::{fmt, marker};

struct LocalType;

// Allowed in Rust Unchained.
impl fmt::Display for *mut LocalType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "This not compile")
    }
}

impl<T> marker::Copy for *mut T {
    //~^ ERROR E0119
	//~| NOTE conflicting implementation in crate `core`
}

fn main() {}
