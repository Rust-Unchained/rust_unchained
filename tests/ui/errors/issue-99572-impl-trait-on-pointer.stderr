error[E0119]: conflicting implementations of trait `Copy` for type `*mut _`
  --> $DIR/issue-99572-impl-trait-on-pointer.rs:14:1
   |
LL | impl<T> marker::Copy for *mut T {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T> Copy for *mut T
             where T: ?Sized;

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
